presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -58,6 +58,11 @@
     Slice getSlice(int position, int offset, int length);
 
     /**
+     * Gets an object in the value at {@code position}.
+     */
+    public <T> T getObject(int position, Class<T> clazz);
```

Comment:
`public` is redundant on an interface.

Maybe add a default implementation that throws an exception